internal/pkg/term/selector/selector.go

```diff
@@ -334,7 +353,18 @@ func (s *ConfigSelect) retrieveServices(app string) ([]string, error) {
 }
 
 func (s *WorkspaceSelect) retrieveWorkspaceServices() ([]string, error) {
-	localServiceNames, err := s.svcLister.ServiceNames()
+	localServiceNames, err := s.wlLister.ServiceNames()
+	if err != nil {
+		return nil, err
+	}
+	if len(localServiceNames) == 0 {
+		return nil, errors.New("no services found in workspace")
+	}
```

Comment:
I think we should let the caller of `retrieveWorkspaceServices()` decide whether empty list means an error. Is this always the case for all callers atm? Otherwise, we might be introducing a breaking change.

Reply:
Both `retrieve` methods now return empty lists if they don't find any jobs/services. `Service()` and `Job()` now generate the error.